Sources/Plasma/Apps/plClient/plClient.cpp

```diff
@@ -413,7 +419,18 @@ void plClient::ISetGraphicsDefaults()
     plDynamicCamMap::SetEnabled(plPipeline::fDefaultPipeParams.PlanarReflections ? true : false);
 }
 
-bool plClient::InitPipeline()
+plPipeline* plClient::ICreatePipeline(hsWindowHndl disp, hsWindowHndl hWnd, const hsG3DDeviceModeRecord* devMode)
+{
+#if defined(PLASMA_PIPELINE_DX)
+    return new plDXPipeline(hWnd, devMode);
+#elif defined(PLASMA_PIPELINE_GL)
+    return new plGLPipeline(disp, hWnd, devMode);
+#else
+    return nullptr;
```

Comment:
This isn't a suggestion for you to change anything in this PR, but it might be interesting to have a `plNullPipeline` so that the work on developing a cross-platform rendering pipeline and a cross-platform client can be somewhat decoupled. There are a few things that can be done, I think, without the actual rendering code.

Reply:
Added a `plNullPipeline` implementation